### PR TITLE
Auth theme

### PR DESCRIPTION
**Pull request by JAYBORICHA07**
- auth fix.
- implemented theme in way ( can toggle between light and dark).
- adding tailwind.
- new router in frontend.

### CODE REVIEW

**Review by KMJ-007 — APPROVED** [review]

File: apps/api/src/app.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ app.setErrorHandler(function (error, _request, reply) {
 });
 
 app.register(cookiePlugin);
-app.register(fastifyJwt, { secret: env.JWT_SECRET });
+app.register(fastifyJwt, { secret: env.JWT_SECRET as string });
```
Comment (krishna-404) [inline apps/api/src/app.ts:39]: why do you have to use `as` here? `env` is already giving this type defined right?

File: apps/api/src/app.ts [hunk 2]
```diff
@@ -46,8 +46,8 @@ app.register(oauthPlugin, {
   credentials: {
     auth: oauthPlugin.GOOGLE_CONFIGURATION,
     client: {
-      id: env.GOOGLE_CLIENT_ID,
-      secret: env.GOOGLE_CLIENT_SECRET,
+      id: env.GOOGLE_CLIENT_ID as string,
+      secret: env.GOOGLE_CLIENT_SECRET as string,
```
Comment (krishna-404) [inline apps/api/src/app.ts:50]: why do you have to use `as` here? `env` is already giving this type defined right?

File: apps/api/src/auth.router.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export const authRouter = (
 
     reply.clearCookie("session", {
       path: "/",
-      domain: new URL(env.FRONTEND_URL).hostname,
+      domain: new URL(env.FRONTEND_URL as string).hostname,
```
Comment (krishna-404) [inline apps/api/src/auth.router.ts:31]: why do you have to use `as` here? `env` is already giving this type defined right?

No point in using typescript if you are going to force the types everywhere.

File: apps/api/src/auth.router.ts [hunk 4]
```diff
@@ -87,15 +87,15 @@ export const authRouter = (
           secure: true,
           sameSite: "none",
           path: "/",
-          domain: new URL(env.FRONTEND_URL).hostname,
+          domain: new URL(env.FRONTEND_URL as string).hostname,
```
Comment (krishna-404) [inline apps/api/src/auth.router.ts:90]: same as above

File: apps/api/src/db/db.config.ts [hunk 5]
```diff
@@ -5,11 +5,11 @@ import { UserTable } from "./tables/user.table";
 export const db = orchidORM(
   {
     // biome-ignore lint/style/useNamingConvention: <as needed by library>
-    databaseURL: env.DB_TEST_URL ?? env.DB_URL,
+    databaseURL: (env.DB_TEST_URL as string) ?? (env.DB_URL as string),
```
Comment (krishna-404) [inline apps/api/src/db/db.config.ts:8]: same as above

File: apps/pwa/package.json
Comment (krishna-404) [inline apps/pwa/package.json:1]: Too many new libs added. Is this well thought & needed?

File: apps/api/src/server.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ const port = Number(process.env.PORT) || 3000;
 
 // TODO: Figure out how to get secure-session https://www.npmjs.com/package/@fastify/secure-session
 server.register(fastifyCors, {
-  origin: [env.FRONTEND_URL],
+  origin: [env.FRONTEND_URL as string],
```
Comment (krishna-404) [inline apps/api/src/server.ts:13]: same as above

File: apps/pwa/src/App.tsx [hunk 7]
```diff
@@ -1,34 +1,24 @@
...
-              </trpc.Provider>
-            </MotionLazyContainer>
-          </BrowserRouter>
-        </CollapseDrawerProvider>
-      </AntdThemeProvider>
-    </HelmetProvider>
+    <AntdConfig>
+      <AntdApp>
+        <MotionLazy>
+          <Helmet>
```
Comment (krishna-404) [inline apps/pwa/src/App.tsx:14]: What is difference between `Helmet` & `HelmetProvider`?

File: apps/pwa/src/assets/icons/ic-analysis.svg
Comment (krishna-404) [inline apps/pwa/src/assets/icons/ic-analysis.svg:1]: When we are already importing `@iconify/react`, why are we also hardcoding the svgs?

File: apps/pwa/src/components/animate/motion-container.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+ *   hidden: { opacity: 0 },
+ * }
+ *
+ * 需要指定 inital 和 animate 属性名
+ * <motion.div
+ *  initial="hidden"
+ *  animate="visible"
+ *  variants={variants}
+ * />
+ */
```
Comment (krishna-404) [inline apps/pwa/src/components/animate/motion-container.tsx:26]: Why do we have this?

File: apps/pwa/src/components/animate/motion-viewport.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+interface Props extends MotionProps {
+  className?: string;
+}
+/**
+ * [whileInView: 元素可以在进出视口时设置动画](https://www.framer.com/motion/scroll-animations/#scroll-triggered-animations)
+ *
+ * + viewport: [视口](https://www.framer.com/motion/scroll-animations/###viewport)
+ *
+ *    + once: 仅触发一次
+ */
```
Comment (krishna-404) [inline apps/pwa/src/components/animate/motion-viewport.tsx:14]: Lets have stuff in english or not at all please.

File: apps/pwa/src/components/animate/types.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+export type VariantsType = {
```
Comment (krishna-404) [inline apps/pwa/src/components/animate/types.ts:1]: Variants of what? also no need to suffix `Type` everywhere.

**Pull request merged**